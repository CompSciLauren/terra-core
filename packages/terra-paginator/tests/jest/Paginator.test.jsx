import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { mountWithIntl } from 'terra-enzyme-intl';

import Paginator from '../../src/Paginator';

describe('Paginator', () => {
  // Snapshot Tests
  it('should render a default paginator', () => {
    const wrapper = mountWithIntl(<Paginator onPageChange={e => typeof e} selectedPage={1} totalCount={2234} itemCountPerPage={20} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a paginator without page links', () => {
    const wrapper = mountWithIntl(<Paginator onPageChange={e => typeof e} />);
    expect(wrapper).toMatchSnapshot();
  });
});
