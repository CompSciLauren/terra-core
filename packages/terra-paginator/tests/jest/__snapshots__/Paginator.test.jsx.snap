// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator should render a default paginator 1`] = `
<ResponsiveElement
  medium={
    <div
      className="paginator"
    >
      <PaginatorButton
        aria-disabled={true}
        aria-label="Terra.paginator.first"
        className="nav-link left-controls is-disabled"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={null}
        type="button"
      >
        Terra.paginator.first
      </PaginatorButton>
      <PaginatorButton
        aria-disabled={true}
        aria-label="Terra.paginator.previous"
        className="nav-link left-controls previous is-disabled"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={null}
        type="button"
      >
        <span
          className="icon"
        />
        Terra.paginator.previous
      </PaginatorButton>
      <PaginatorButton
        aria-current="page"
        aria-label="Terra.paginator.pageIndex"
        className="nav-link is-selected"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={null}
        type="button"
      >
        1
      </PaginatorButton>
      <PaginatorButton
        aria-current={false}
        aria-label="Terra.paginator.pageIndex"
        className="nav-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        2
      </PaginatorButton>
      <PaginatorButton
        aria-current={false}
        aria-label="Terra.paginator.pageIndex"
        className="nav-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        3
      </PaginatorButton>
      <PaginatorButton
        aria-current={false}
        aria-label="Terra.paginator.pageIndex"
        className="nav-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        4
      </PaginatorButton>
      <PaginatorButton
        aria-current={false}
        aria-label="Terra.paginator.pageIndex"
        className="nav-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        5
      </PaginatorButton>
      <PaginatorButton
        aria-current={false}
        aria-label="Terra.paginator.pageIndex"
        className="nav-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        6
      </PaginatorButton>
      <PaginatorButton
        aria-current={false}
        aria-label="Terra.paginator.pageIndex"
        className="nav-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        7
      </PaginatorButton>
      <PaginatorButton
        aria-current={false}
        aria-label="Terra.paginator.pageIndex"
        className="nav-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        8
      </PaginatorButton>
      <PaginatorButton
        aria-current={false}
        aria-label="Terra.paginator.pageIndex"
        className="nav-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        9
      </PaginatorButton>
      <PaginatorButton
        aria-current={false}
        aria-label="Terra.paginator.pageIndex"
        className="nav-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        10
      </PaginatorButton>
      <PaginatorButton
        aria-disabled={false}
        aria-label="Terra.paginator.next"
        className="nav-link right-controls next"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        Terra.paginator.next
        <span
          className="icon"
        />
      </PaginatorButton>
      <PaginatorButton
        aria-disabled={false}
        aria-label="Terra.paginator.last"
        className="nav-link right-controls"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        Terra.paginator.last
      </PaginatorButton>
    </div>
  }
  responsiveTo="parent"
  tiny={
    <div
      aria-label="pagination"
      className="paginator"
      role="navigation"
    >
      <PaginatorButton
        aria-disabled={true}
        aria-label="Terra.paginator.first"
        className="nav-link left-controls is-disabled"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={null}
        type="button"
      >
        Terra.paginator.first
      </PaginatorButton>
      <PaginatorButton
        aria-disabled={true}
        aria-label="Terra.paginator.previous"
        className="nav-link left-controls previous icon-only is-disabled"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={null}
        type="button"
      >
        <VisuallyHiddenText
          text="Terra.paginator.previous"
        />
        <span
          className="icon"
        />
      </PaginatorButton>
      Terra.paginator.pageIndex
      <PaginatorButton
        aria-disabled={false}
        aria-label="Terra.paginator.next"
        className="nav-link right-controls next icon-only"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        <VisuallyHiddenText
          text="Terra.paginator.next"
        />
        <span
          className="icon"
        />
      </PaginatorButton>
      <PaginatorButton
        aria-disabled={false}
        aria-label="Terra.paginator.last"
        className="nav-link right-controls"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        Terra.paginator.last
      </PaginatorButton>
    </div>
  }
/>
`;

exports[`Paginator should render a paginator without page links 1`] = `
<ResponsiveElement
  medium={
    <div
      className="paginator pageless"
    >
      <PaginatorButton
        aria-disabled={false}
        aria-label="Terra.paginator.previous"
        className="nav-link left-controls previous"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="icon"
        />
        Terra.paginator.previous
      </PaginatorButton>
      <PaginatorButton
        aria-disabled={false}
        aria-label="Terra.paginator.next"
        className="nav-link right-controls next"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        Terra.paginator.next
        <span
          className="icon"
        />
      </PaginatorButton>
    </div>
  }
  responsiveTo="parent"
  tiny={
    <div
      aria-label="pagination"
      className="paginator pageless"
      role="navigation"
    >
      <PaginatorButton
        aria-disabled={false}
        aria-label="Terra.paginator.previous"
        className="nav-link left-controls previous icon-only"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        <VisuallyHiddenText
          text="Terra.paginator.previous"
        />
        <span
          className="icon"
        />
      </PaginatorButton>
      <PaginatorButton
        aria-disabled={false}
        aria-label="Terra.paginator.next"
        className="nav-link right-controls next icon-only"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        <VisuallyHiddenText
          text="Terra.paginator.next"
        />
        <span
          className="icon"
        />
      </PaginatorButton>
    </div>
  }
/>
`;
